docs/api.yaml

```diff
@@ -0,0 +1,1061 @@
+openapi: "3.0.3"
+info:
+  #  description:
+
+  version: "1.0.0"
+  title: "HEI Admin API"
+  contact:
+    email: "contact@hei.school"
+
+  #servers:
+  # - description: To complete
+  #   url: example.com
+
+paths:
+  /groups:
+    get:
+      tags:
+        - "Groups"
+      summary: "Find groups based on search criteria"
+      parameters:
+        - name: "id"
+          in: "query"
+          description: "Identifier of the group to find"
+          required: false
+          schema:
+            type: string
+        - name: offset
+          in: query
+          required: false
+          description: "Zero-based offset for pagination. Defaults to `0`."
+          schema:
+            type: integer
+        - name: limit
+          in: query
+          required: false
+          description: "Pagination limit, that is the maximum number of elements that can be retrieved. Defaults to `1000` and cannot exceed `1000`."
+          schema:
+            type: integer
+
+      operationId: "findGroups"
+      responses:
+        200:
+          description: "List of groups"
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Group'
+        400:
+          $ref: "#/components/responses/400"
+        403:
+          $ref: "#/components/responses/403"
+        404:
+          $ref: "#/components/responses/404"
+        429:
+          $ref: "#/components/responses/429"
+        500:
+          $ref: "#/components/responses/500"
+
+    post:
+      tags:
+        - "Groups"
+      summary: "Create new groups"
+      description: ""
+      operationId: "createGroups"
+      requestBody:
+        description: "The collection of groups to create"
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateGroup'
+      responses:
+        200:
+          description: "Collection of created groups"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Group'
+        400:
+          $ref: "#/components/responses/400"
+        403:
+          $ref: "#/components/responses/403"
+        404:
+          $ref: "#/components/responses/404"
+        429:
+          $ref: "#/components/responses/429"
+        500:
+          $ref: "#/components/responses/500"
+
+    patch:
+      tags:
+        - "Groups"
+      summary: "Update an existing group"
+      parameters:
```

Comment:
Add a (ref: string) parameter, described as follows: "Human-readable identifier for the resource". Typically, ref is auto-incrementable field which will be shown afterwards on all user interfaces.

On the other hand, id is an uuid, which is NOT auto-increment-able. As search are based on id, we MUST make it non-autoincrementable (so that malicious people cannot guess our range of id).

Please also add ref to all other resources. Note that ref is NOT updatable.

Reply:
Is there a specific or general standard to write this ref according to the alphabetic and numeric characters ?
Or this detail isn't important in Rest API ?

If it is : 
How many alphabetic chars ? How do we "split" it ? Do we use underscore or not ? How many digits for the numeric part ?
For example for student : "STD_00001" or "STD000001".

Reply:
No standard way, it's actually business-dependent